openingd/dualopend.c

```diff
@@ -596,28 +597,31 @@ static size_t psbt_input_weight(struct wally_psbt *psbt,
 				size_t in)
 {
 	size_t weight;
+	const struct wally_map_item *redeem_script = wally_map_get_integer(&psbt->inputs[in].psbt_fields, /* PSBT_IN_REDEEM_SCRIPT */ 0x04);
 
 	/* txid + txout + sequence */
 	weight = (32 + 4 + 4) * 4;
-	weight +=
-		(psbt->inputs[in].redeem_script_len +
-			(varint_t) varint_size(psbt->inputs[in].redeem_script_len)) * 4;
+	if (redeem_script) {
+		weight +=
+			(redeem_script->value_len +
+				(varint_t) varint_size(redeem_script->value_len)) * 4;
+	}
```

Comment:
If we don't have a redeem_script, is that weight 0 (as here) or weight 1 (as before)?

Reply:
nice catch